cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/keyspaces/KeyspaceCreator.java

```diff
@@ -89,9 +90,31 @@ public void createKeyspace() throws IOException, CradleStorageException
 				: SchemaBuilder.createKeyspace(keyspace).withSimpleStrategy(settings.getKeyspaceReplicationFactor());
 		queryExecutor.executeQuery(createKs.asCql(), true);
 		logger.info("Keyspace '{}' has been created", keyspace);
-		this.keyspaceMetadata = obtainKeyspaceMetadata().get();  //FIXME: keyspace creation may take time and it won't be available immediately
+
+		awaitKeyspaceReady();
 	}
-	
+
+	private void awaitKeyspaceReady() throws IOException
+	{
+		int timeout = 500;
+		int attempt = 0;
+		for (;getKeyspaceMetadata() == null && attempt < 5; attempt++)
+		{
+			try
+			{
+				TimeUnit.MILLISECONDS.sleep(timeout);
+			}
+			catch (InterruptedException e)
+			{
+				throw new IOException("Awaiting of keyspace ready '"+keyspace+"' has been interrupted");
```

Comment:
Replace message with "Waiting for keyspace '"+keyspace+"' readiness has been interrupted".
And it is not IOException (not related to input-output).

Reply:
Will be it ok if it will be CradleStorageException instead IOException?

Reply:
Yes.